lib/hyperclient/collection.rb

```diff
@@ -10,6 +10,8 @@ module Hyperclient
   class Collection
     include Enumerable
 
+    undef :filter
```

Comment:
I'm not happy with this, but it's the only way I found to keep the compatibility. The alternative is removing the magic that allows accessing properties via `links.property`, and force devs to write `links[:property]` instead.

Thoughts on this? 😄